### ISSUE TITLE
False positive when using a constrained TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2492]
### Discussed in https://github.com/microsoft/pyright/discussions/2489

<div type='discussions-op-text'>

<sup>Originally posted by **jmg-duarte** October 26, 2021</sup>
Hello all, 

I have the following code:
```python
from typing import Type, TypeVar

T = TypeVar("T", int, bool)

def foo(v: T, t: Type[T]):
    print(t)

def bar(v: T, t: Type[T]):
    foo(v, t)
```

Pyright/Pylance report an error on the last line, on top of `t`:
```
Argument of type "Type[T@bar]" cannot be assigned to parameter "t" of type "Type[T@foo]" in function "foo"
  Type "Type[T@bar]" is incompatible with constrained type variable "T@bar"
```
